test/tools/router_check/router.cc

```diff
@@ -186,7 +175,7 @@ bool RouterCheckTool::compareEntries(const std::string& expected_routes) {
     tool_config.route_ = config_->route(*tool_config.headers_, tool_config.random_value_);
 
     const std::string& test_name = check_config.test_name();
-    tests_.push_back(std::pair<std::string, std::vector<std::string>>(test_name, {}));
+    tests_.emplace_back(std::pair<std::string, std::vector<std::string>>(test_name, {}));
```

Comment:
did `tests_.emplace_back(test_name, {});` not work?

Reply:
No, I don't think emplace_back can infer the type.

Reply:
Try `tests_.emplace_back(std::make_pair(test_name, {}));` If that doesn't work we can drop it.

Reply:
It seems that emplace_back/make_pair can't infer the type of the empty vector even though it's being added to tests_. So ``tests_.emplace_back(std::make_pair(test_name,  std::vector<std::string>{}));`` works, but only because the type of the empty vector is explicit.

Reply:
I'll use ``tests_.emplace_back(test_name, std::vector<std::string>{});``